extensions/arc/src/ui/components/radioOptionsGroup.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { getErrorMessage } from '../../common/utils';
+
+export interface RadioOptionsInfo {
+	values?: string[],
+	defaultValue: string
+}
+
+export class RadioOptionsGroup {
+	private _divContainer!: azdata.DivContainer;
+	private _loadingBuilder: azdata.LoadingComponentBuilder;
+	private _currentRadioOption!: azdata.RadioButtonComponent;
+	constructor(private _view: azdata.ModelView, private _onNewDisposableCreated: (disposable: vscode.Disposable) => void) {
+		this._divContainer = this._view!.modelBuilder.divContainer().withProperties<azdata.DivContainerProperties>({ clickable: false }).component();
+		this._loadingBuilder = this._view!.modelBuilder.loadingComponent().withItem(this._divContainer);
+	}
+
+	public component(): azdata.LoadingComponent {
+		return this._loadingBuilder.component();
+	}
+
+	async load(optionsInfoGetter: () => Promise<RadioOptionsInfo>): Promise<void> {
+		this.component().loading = true;
+		this._divContainer.clearItems();
+		try {
+			const optionsInfo = await optionsInfoGetter();
+			const options = optionsInfo.values!;
+			let defaultValue: string = optionsInfo.defaultValue!;
+			options.forEach((option: string) => {
+				const radioOption = this._view!.modelBuilder.radioButton().withProperties<azdata.RadioButtonProperties>({
+					label: option,
+					checked: option === defaultValue,
+					name: option,
+					value: option,
+					enabled: true
+				}).component();
+				if (radioOption.checked) {
+					this._currentRadioOption = radioOption;
+				}
+				this._onNewDisposableCreated(radioOption.onDidClick(() => {
```

Comment:
Each radioButton object is inpdependent of other, so I do not know/see what could be un-checking radio button on the previously checked button when a new one is checked. Is there a RadioButtonGroup object now shared through azdata that I should be using instead of individual radio button?

Reply:
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_radio

Note that the state of the radio buttons are done as a unit - this is functionality provided by using the "name" property to group the buttons together. 

But I don't think we currently utilize this currently.

The ideal behavior - the ModelView components understand this and hook into the events to properly update the components with their enabled state as the buttons are selected/unselected. This is what I was referring to with my first comment. 

I seem to recall looking at this before though and finding out that there isn't any event hooked up to when a radio button was unselected so it would be pretty difficult to make sure the state for each button is kept correctly updated. 

So it'd be good to verify that yourself now - and at the very least this kind of logic to update the checked property shouldn't be necessary with how it looks like you have this set up since you don't actually care about keeping the checked property up to date. You just care about whatever the last one selected was.

Reply:
Ok will check if it works if I start setting the name property to group the buttons together.

Reply:
Ok removed forEach loop to check all the radio buttons. Checked on your suggestion, if we do not set checked to false of previously checked radio Button, the UI and all functions work fine. It is a little weird that a previously checked radio button continues to store 'checked' as true in that scenario though everything works. While storing this._currentRadioOption, I am not simply unchecking the state of the previous radio option.

Reply:
Sounds good. It would be nice if the state was updated but that should be a thing handled on the main thread side anyways if we do end up handling it I would think so doing it here wasn't the greatest anyways.

Reply:
If you think it is better not to do the update here then I can leave just a comment here without the corresponding update. We can do the right fix on the main thread separately, I do not want to pull that change in here.